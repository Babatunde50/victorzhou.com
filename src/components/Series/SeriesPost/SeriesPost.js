// @flow
import React from 'react';
import styles from './SeriesPost.module.scss';
import ContentDate from '../../ContentDate';

type Props = {|
  +date: string,
  +dateModified: string,
  +description: string,
  +img: string,
  +n: number,
  +slug: string,
  +title: string,
|};

const SeriesPost = ({ date, dateModified, description, img, n, slug, title }: Props) => (
  <div className={styles['series-post']}>
    <a href={slug}>
      <img src={img} />
    </a>
    <div>
      <h3>
        <a href={slug}>{`${n}. ${title}`}</a>
      </h3>
      <ContentDate date={date} dateModified={dateModified} />
      <p>{description}</p>
    </div>
  </div>
);

export default SeriesPost;
