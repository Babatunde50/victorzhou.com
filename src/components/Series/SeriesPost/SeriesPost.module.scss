@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.series-post {
  padding: 10px;
  margin: 10px 0;
  background-color: $color-gray-bg;
  border-radius: 5px;

  img {
    width: 100%;
    margin: 0 auto;
    border-radius: 3px;
  }

  h3 {
    margin: 10px 0 5px 0;
  }

  p {
    margin: 0;
  }
}

@include breakpoint-md {
  .series-post {
    padding: 15px;
    margin: 10px 20px;

    p {
      margin: 5px 0 0 0;
    }
  }
}
