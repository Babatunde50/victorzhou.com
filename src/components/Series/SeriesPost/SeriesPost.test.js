import React from 'react';
import renderer from 'react-test-renderer';
import SeriesPost from './SeriesPost';

describe('SeriesPost', () => {
  const props = {
    date: new Date('2019-03-14T12:00:00.000Z'),
    description: 'test-description',
    img: 'test-img',
    n: 1,
    slug: 'test-slug',
    title: 'test-title',
  };

  it('renders correctly', () => {
    const tree = renderer.create(<SeriesPost {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
