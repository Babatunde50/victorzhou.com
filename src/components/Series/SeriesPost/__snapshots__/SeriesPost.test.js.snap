// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesPost renders correctly 1`] = `
<div
  className="series-post"
>
  <a
    href="test-slug"
  >
    <img
      src="test-img"
    />
  </a>
  <div>
    <h3>
      <a
        href="test-slug"
      >
        1. test-title
      </a>
    </h3>
    <p
      className="content-date"
    >
      <time>
        July 29, 2019
      </time>
    </p>
    <p>
      test-description
    </p>
  </div>
</div>
`;
